pkg/ingester/stream.go

```diff
@@ -309,14 +321,15 @@ func (s *stream) Push(
 
 	if len(failedEntriesWithError) > 0 {
 		lastEntryWithErr := failedEntriesWithError[len(failedEntriesWithError)-1]
-		if lastEntryWithErr.e != chunkenc.ErrOutOfOrder && lastEntryWithErr.e != ErrStreamRateLimit {
+		_, ok := lastEntryWithErr.e.(*ErrStreamRateLimit)
```

Comment:
What's going on here?

Reply:
The error is a struct that satisfies the `Error` interface now, we have to typecheck instead of just comparing the concrete errors directly.